command/exportedservices/exported_services.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package exportedservices
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/cli"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/command/flags"
+	"github.com/ryanuber/columnize"
+)
+
+const (
+	PrettyFormat string = "pretty"
+	JSONFormat   string = "json"
+)
+
+func getSupportedFormats() []string {
+	return []string{PrettyFormat, JSONFormat}
+}
+
+func formatIsValid(f string) bool {
+	return f == PrettyFormat || f == JSONFormat
+}
+
+func New(ui cli.Ui) *cmd {
+	c := &cmd{UI: ui}
+	c.init()
+	return c
+}
+
+type cmd struct {
+	UI    cli.Ui
+	flags *flag.FlagSet
+	http  *flags.HTTPFlags
+	help  string
+
+	format string
+}
+
+func (c *cmd) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	c.flags.StringVar(
+		&c.format,
+		"format",
+		PrettyFormat,
+		fmt.Sprintf("Output format {%s} (default: %s)", strings.Join(getSupportedFormats(), "|"), PrettyFormat),
+	)
+
+	c.http = &flags.HTTPFlags{}
+	flags.Merge(c.flags, c.http.ClientFlags())
+	flags.Merge(c.flags, c.http.ServerFlags())
+	flags.Merge(c.flags, c.http.PartitionFlag())
+	c.help = flags.Usage(help, c.flags)
+}
+
+func (c *cmd) Run(args []string) int {
+	if err := c.flags.Parse(args); err != nil {
+		return 1
+	}
+
+	if !formatIsValid(c.format) {
+		c.UI.Error(fmt.Sprintf("Invalid format, valid formats are {%s}", strings.Join(getSupportedFormats(), "|")))
+		return 1
+	}
+
+	client, err := c.http.APIClient()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error connect to Consul agent: %s", err))
+		return 1
+	}
+
+	exportedServices, _, err := client.ExportedServices(nil)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error reading exported services: %v", err))
+		return 1
+	}
+
+	if len(exportedServices) == 0 {
+		c.UI.Info("No exported services found")
+		return 0
+	}
+
+	if c.format == JSONFormat {
+		output, err := json.MarshalIndent(exportedServices, "", "    ")
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error marshalling JSON: %s", err))
+			return 1
+		}
+		c.UI.Output(string(output))
+		return 0
+	}
+
+	c.UI.Output(formatExportedServices(exportedServices))
+
+	return 0
+}
+
+func formatExportedServices(services []api.ResolvedExportedService) string {
+	result := make([]string, 0, len(services)+1)
```

Comment:
This needs to be fixed. I am getting this response 

```
asheshvidyut@asheshvidyut-H2GX766V9T ~/consul (tauhid621_exported_services_docs_and_cli *) » ./bin/consul exported-services
Service  Consumer
test     Peer: test
test     Partition: test
test1    test2              Peer: test1
test1    test2              Peer: batman
test1    test2              Peer: superman
test1    test2              Partition: test2
test1    test2              Partition: heman
test1    test2              Partition: spiderman
```

Reply:
I think it's because of mixing of enterprise and non enterprise response. In case of enterprise all the services will have partition and in case of CE partition will be `""` string.

Reply:
The enterprise features partition and namespace are going to change. (from RFC which I shared with you on slack).
You can take a call based on that.

Reply:
I think it will be better to keep all four columsn.

Reply:
Got it my bad.